app/models/user_project_role.rb

```diff
@@ -0,0 +1,15 @@
+class UserProjectRole < ActiveRecord::Base
```

Comment:
has_and_belongs_many should be avoided if possible, it locks you into some of the worst rails magic.